packages/xml-views-validation/test/validators/element/non-stable-id-spec.ts

```diff
@@ -0,0 +1,172 @@
+
+describe("the use of non stable id validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({
+      version: "1.74.0",
+      modelGenerator: generate,
+    });
+  });
+
+  context("true positive scenarios", () => {
+    let assertSingleIssue: (xmlSnippet: string, message: string) => void;
+    before(() => {
+      assertSingleIssue = partial(
+        assertSingleIssueBase,
+        ui5SemanticModel,
+        {
+          element: [validateNonStableId],
+        },
+        "NonStableIDIssue",
+        "error"
+      );
+    });
+
+    it("will detect missing stable id in non-whitelisted UI5 class", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons">
+            <🢂m:Panel🢀>
+            </m:Panel>
+        </mvc:View>`,
+        getMessage(NON_STABLE_ID, "Panel")
+      );
+    });
+
+    it("will detect missing stable id in custom control", () => {
+      assertSingleIssue(
+        `<🢂custom:View🢀
+          xmlns:custom="foo.bar"
+          xmlns="bar.foo">
+            <Button id="dummy-id">
+            </Button>
+        </custom:View>`,
+        getMessage(NON_STABLE_ID, "View")
+      );
+    });
+
+    it("will detect missing stable id in whitelisted root class when it's not in the root", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.ui.commons">
+              <🢂m:Panel🢀>
+              </m:Panel>
+          </mvc:View>`,
+        getMessage(NON_STABLE_ID, "Panel")
+      );
+    });
+
+    it("will detect missing stable in sub element when the element has attribute sap.ui.dt:designtime='not-adaptable'", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons"
+          xmlns:sap.ui.dt="sap.ui.dt">
+            <m:Panel sap.ui.dt:designtime="not-adaptable">
+              <🢂m:Button🢀></m:Button>
+            </m:Panel>
+        </mvc:View>`,
+        getMessage(NON_STABLE_ID, "Button")
+      );
+    });
+  });
+
+  context("negative edge cases", () => {
+    let assertNoIssues: (xmlSnippet: string) => void;
+    before(() => {
+      assertNoIssues = partial(assertNoIssuesBase, ui5SemanticModel, {
+        element: [validateNonStableId],
+      });
+    });
+
+    it("will not detect an issue when there is a root-whitelisted UI5 class", () => {
+      assertNoIssues(
+        `<!-- sap.ui.core.mvc.View is whitlisted root class -->
+        <mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons">
+        </mvc:View>`
+      );
+    });
+
+    it("will not detect an issue when there is a root-whitelisted UI5 class - core ns", () => {
```

Comment:
`when there is` -> `for a` because the existence of the whitelisted element
does not mean all other elements would not be checked